voicebox/encryption.py

```diff
@@ -15,37 +15,117 @@
 
 class BaseEncryptor:
 
-    KEY_EXCHANGE_SIGNAL = 900
+    """
+    BaseEncryptor Class
+
+    Base class for encryptors providing methods for encryption, decryption, and hashing.
 
-    def __init__(self, *args, **kwargs):
+    Methods:
+        encrypt: Encrypts the payload (to be implemented by subclasses).
+        decrypt: Decrypts the packet (to be implemented by subclasses).
+        hash: Computes the hash of the payload (to be implemented by subclasses).
+        packet_handler: Default packet handler that returns the packet unchanged.
+        public_pem: Property representing the public PEM key, returns None by default.
 
-        raise NotImplementedError("Cannot Initialize from BaseEncryptor")
+    Attributes:
+        KEY_EXCHANGE_SIGNAL (int): A constant representing the key exchange signal.
+    """
+
+    KEY_EXCHANGE_SIGNAL = 900
 
     def encrypt(self, payload: bytes):
+        """
+        Encrypts the payload.
+
+        Args:
+            payload (bytes): The payload to be encrypted.
+
+        Raises:
+            NotImplementedError: This method should be implemented by subclasses.
+        """
 
         raise NotImplementedError()
 
     def decrypt(self, packet: bytes):
+        """
+        Decrypts the packet.
+
+        Args:
+            packet (bytes): The encrypted packet to be decrypted.
 
+        Raises:
+            NotImplementedError: This method should be implemented by subclasses.
+        """
         raise NotImplementedError()
 
     def hash(self, payload):
+        """
+        Computes the hash of the payload.
 
-        raise NotImplementedError()
+        Args:
+            payload: The payload to be hashed.
+
+        Raises:
+            NotImplementedError: This method should be implemented by subclasses.
+        """
+        raise hash(payload)
```

Comment:
It seems like the `hash` method in the `BaseEncryptor` class is not implemented correctly. It raises the hash of the payload instead of a `NotImplementedError`. This could potentially lead to unexpected behavior. [important]